nixos/modules/services/web-apps/dokuwiki.nix

```diff
@@ -18,244 +19,350 @@ let
+                name = "icalevents";
+                # Download the plugin from the dokuwiki site
+                src = pkgs.fetchurl {
+                  url = https://github.com/real-or-random/dokuwiki-plugin-icalevents/releases/download/2017-06-16/dokuwiki-plugin-icalevents-2017-06-16.zip;
+                  sha256 = "e40ed7dd6bbe7fe3363bbbecb4de481d5e42385b5a0f62f6a6ce6bf3a1f9dfa8";
+                };
+                sourceRoot = ".";
+                # We need unzip to build this package
+                buildInputs = [ pkgs.unzip ];
+                # Installing simply means copying all files to the output directory
+                installPhase = "mkdir -p $out; cp -R * $out/";
+              };
+
+              # And then pass this theme to the plugin list like this:
+              plugins = [ plugin-icalevents ];
+        '';
+      };
+
+      templates = mkOption {
+        type = types.listOf types.path;
+        default = [];
+        description = ''
+              List of path(s) to respective template(s) which are copied from the 'tpl' directory.
+              <note><para>These templates need to be packaged before use, see example.</para></note>
+        '';
+        example = ''
+              # Let's package the bootstrap3 theme
+              template-bootstrap3 = pkgs.stdenv.mkDerivation {
+                name = "bootstrap3";
+                # Download the theme from the dokuwiki site
+                src = pkgs.fetchurl {
+                  url = https://github.com/giterlizzi/dokuwiki-template-bootstrap3/archive/v2019-05-22.zip;
+                  sha256 = "4de5ff31d54dd61bbccaf092c9e74c1af3a4c53e07aa59f60457a8f00cfb23a6";
+                };
+                # We need unzip to build this package
+                buildInputs = [ pkgs.unzip ];
+                # Installing simply means copying all files to the output directory
+                installPhase = "mkdir -p $out; cp -R * $out/";
+              };
+
+              # And then pass this theme to the template list like this:
+              templates = [ template-bootstrap3 ];
+        '';
+      };
+
+      poolConfig = mkOption {
+        type = with types; attrsOf (oneOf [ str int bool ]);
+        default = {
+          "pm" = "dynamic";
+          "pm.max_children" = 32;
+          "pm.start_servers" = 2;
+          "pm.min_spare_servers" = 2;
+          "pm.max_spare_servers" = 4;
+          "pm.max_requests" = 500;
+        };
+        description = ''
+          Options for the dokuwiki PHP pool. See the documentation on <literal>php-fpm.conf</literal>
+          for details on configuration directives.
+        '';
+      };
+
+      nginx = mkOption {
+        type = types.submodule (
+          recursiveUpdate
+            (import ../web-servers/nginx/vhost-options.nix { inherit config lib; })
+            {
+              # Enable encryption by default,
+              options.forceSSL.default = true;
+              options.enableACME.default = true;
```

Comment:
I know I have introduced this option, but on a second thought, it is quite unnecessary to force these settings upon user. Wouldn't it be reasonable to drop it - along with forceSSL?

But in that case, I guess: `fastcgi_param HTTPS on;` would have to go in that case, too.

Reply:
I guess it would be wise to go with the default that works in most cases. ACME does not always work depending on restrictive DNS CA records or missing internet connection.

The [default](https://nixos.org/nixos/options.html#forcessl) seems to be to disable ACME and not to force SSL. But it should be enabled in the example.

Regarding [ninx fastcgi](https://nginx.org/en/docs/http/ngx_http_fastcgi_module.html), if_not_empty should do the trick.

>  If the directive is specified with if_not_empty (1.1.11) then such a parameter will be passed to the server only if its value is not empty:
>
>    fastcgi_param HTTPS           $https if_not_empty;

Reply:
Sounds about right, let's do it that way.

Reply:
Since would be an independent change, so you could open a new MR for this.

Reply:
Sure, could do that, but since you did such a great job in refactoring a few things here and there, I though it would be a nice addition to the PR. But you're right, we can definitely keep it separated, it's just nice to have and not subject of this PR.